ban-cooking.lua

```diff
@@ -0,0 +1,440 @@
+-- convenient way to ban cooking categories of food
+-- based on ban-cooking.rb by Putnam: https://github.com/DFHack/scripts/pull/427/files
+-- Putnams work completed by TBSTeun
+--[====[
+
+ban-cooking
+===========
+A more convenient way to ban cooking various categories of foods than the
+kitchen interface.  Usage:  ``ban-cooking <type>``.  Valid types are ``booze``,
+``honey``, ``tallow``, ``oil``, ``seeds`` (non-tree plants with seeds),
+``brew``, ``fruit``, ``mill``, ``thread``, and ``milk``.
+
+]====]
+
+kitchen = df.global.ui.kitchen
+
+already_banned = {}
+
+local function already_banned_has_key(key)
+    for k, v in pairs(already_banned) do
+        if k[1] == key[1] and k[2] == key[2] and k[3] == key[3] and k[4] == key[4] then
+            return true
+        end
+    end
+
+    return false
+end
```

Comment:
```suggestion
local function make_key(mat_type, mat_index, type, subtype)
    return ('%s:%s:%s:%s'):format(mat_type, mat_index, type, subtype)
end
```

Reply:
I tried to change the key to a different format, but the table as a key is used later, and I am not sure how I can change this key format without losing needed data in the show function